clang/test/Driver/sycl-offload-new-driver.c

```diff
@@ -52,3 +52,16 @@
 // RUN:   | FileCheck -check-prefix WRAPPER_OPTIONS_POSTLINK %s
 // WRAPPER_OPTIONS_POSTLINK: clang-linker-wrapper{{.*}} "--triple=spir64"
 // WRAPPER_OPTIONS_POSTLINK-SAME: "--sycl-post-link-options=-post-link-opt -O2 -spec-const=native -device-globals -split=auto -emit-only-kernels-as-entry-points -emit-param-info -symbols -emit-exported-symbols -split-esimd -lower-esimd"
+
+// -fsycl-device-only behavior
+// RUN: %clangxx --target=x86_64-unknown-linux-gnu -fsycl --offload-new-driver \
+// RUN:          -fsycl-device-only -ccc-print-phases %s 2>&1 \
+// RUN   | FileCheck -check-prefix DEVICE_ONLY %s
+// RUN: %clangxx --target=x86_64-unknown-linux-gnu -fsycl --offload-new-driver \
+// RUN:          --offload-device-only -ccc-print-phases %s 2>&1 \
+// RUN:  | FileCheck -check-prefix DEVICE_ONLY %s
+// DEVICE_ONLY: 0: input, "{{.*}}", c++, (device-sycl)
+// DEVICE_ONLY: 1: preprocessor, {0}, c++-cpp-output, (device-sycl)
+// DEVICE_ONLY: 2: compiler, {1}, ir, (device-sycl)
+// DEVICE_ONLY: 3: backend, {2}, ir, (device-sycl)
```

Comment:
I do not see this phase in the old offloading model. What does this phase do? Thanks

Reply:
I kind of went back and forth on this one, as the new offloading model area implies both of these steps (compiler, which creates the IR, then backend which does processing after that - typically generating assembly).  The assembly generation step is redundant here, as it is combined with the compiler step so we don't actually see both of these phases unless it is purposefully split out.  I think I'll look at this a bit more and just do the compiler step, as it is a cleaner aspect of what we are actually doing.

Reply:
After additional investigation, the additional `backend` step works best with the new offload model path - especially when dealing with being able to emit textual IR with `-fsycl-device-only`.  I will be leaving the change as-is.